### PR TITLE
Add a sass-lint ruleset

### PR DESCRIPTION
**Pull request by marlowpayne**
Status: **Ready for Review**
Owner: @marlowpayne
Reviewers: @jeffkamo @mikenikles @jansepar 

## Changes
- Like #90, but even better because the feature diff seems to be pretty minimal
- This change is in tandem with this generator PR: https://github.com/mobify/generator-adaptivejs/pull/344 , which migrates SCSS linting to [sass-lint](https://github.com/sasstools/sass-lint), finally breaking our dependency on Ruby
- This ruleset needs reviewing/triaging against our existing SCSS linting ruleset to effectively enforce our SCSS code style
- This ruleset was generated using sass-lint's recommended tool for existing SCSS-lint users: http://sasstools.github.io/make-sass-lint-config/

## Todos:
- [x] Evaluate if this is a worth while change
- [x] UI Dev +1
- [ ] Cut a new release of code-style so that the generator no longer needs to point to this branch

### Feedback:
_none so far_

## How to Test
- Checkout the generator PR! https://github.com/mobify/generator-adaptivejs/pull/344

### CODE REVIEW

**Comment by jeffkamo** [comment]
I've compared the Sass-Lint features against SCSS-Lint. See [that comparison here](https://docs.google.com/spreadsheets/d/1ZkMj_wzzPKvpMv7KjeECe3nkoP6qCqkDBKimTOL9KYg/edit#gid=761150116).

File: css/.sass-lint.yml [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+# DisableLinterReason, ElsePlacement, PropertyCount, SelectorDepth
+# SpaceAroundOperator, TrailingWhitespace, UnnecessaryParentReference, VendorPrefixes
+# Compass::*
+#
+# The following settings/values are unsupported by sass-lint:
+# Linter Indentation, option "allow_non_nested_indentation"
+# Linter Indentation, option "character"
+# Linter SpaceBeforeBrace, option "allow_single_line_padding"
+# Linter PropertySortOrder, option "min_properties"
+# Linter PropertySortOrder, option "separate_groups"
```
Comment (jeffkamo) [inline css/.sass-lint.yml]: We can probably remove the above comments. Nice to know, but we have this information recorded elsewhere.

File: css/.sass-lint.yml [hunk 2]
```diff
@@ -0,0 +1,350 @@
...
+        - animation-iteration-count
+        - animation-direction
+        - animation-fill-mode
+        - animation-play-state
+        - null
+        - cursor
+        - user-select
+        - resize
+        - pointer-events
+        - null
```
Comment (jeffkamo) [inline css/.sass-lint.yml]: I suspect these Null values will need to be removed. I'm fairly certain they don't add spaces between properties groups.

Comment (marlowpayne) [inline css/.sass-lint.yml]: Will do. sass-lint won't support separating groups anyways.

File: css/.sass-lint.yml [hunk 3]
```diff
@@ -0,0 +1,350 @@
...
+  trailing-semicolon: 2
+  url-quotes: 2
+  variable-for-property:
+    - 2
+    - properties:
+        - font
+  variable-name-format:
+    - 2
+    - allow-leading-underscore: true
+      convention: '[a-z]|-{3}|_{3}|[^_]_[^_]'
```
Comment (jeffkamo) [inline css/.sass-lint.yml]: Have we tested that these three usages of regex work with our convention?

Comment (marlowpayne) [inline css/.sass-lint.yml]: @jeffkamo this regex is taken from our SCSS-lint `NameFormat` [rule](https://github.com/mobify/mobify-code-style/blob/develop/css/.scss-lint.yml#L132) which is actually lifted from the [SCSS-lint source code](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/selector_format.rb#L58). It's the same regex that SCSS-lint uses for hyphenated BEM format.

**Comment by jeffkamo** [comment]
:+1:

**Comment by mikenikles** [comment]
That looks good to me as well :+1:

**Pull request merged**